### PR TITLE
OCTT - EVSEType representation

### PR DESCRIPTION
**Pull request by deye9**
OCTT - EVSEType representation alongside needed Redux Bug fixes. Fixed the issue where on opening the Set Variables action for a charging station and selecting a Component or Variable, 'EvseDatabaseId is shown when it should show EVSEID and ConnectorID

### CODE REVIEW

File: src/components/data-model-table/association-selection.tsx [hunk 1]
```diff
@@ -49,38 +51,53 @@ export const AssociationSelection = <
...
+      Reflect.getMetadata(
+        PRIMARY_KEY_FIELD_NAME,
+        associatedRecordClassInstance as object,
+      ),
+    [associatedRecordClassInstance],
   );
 
+  const selectedItems = useSelector(getSelectedAssociatedItems(storageKey));
+
+  const selectedIdentifiers = useMemo(() => {
```
Comment (elliot-sabitov) [inline src/components/data-model-table/association-selection.tsx:91]: I like that you moved this .join into here, I am not sure if we will be joining it like this in many other places, so having it like this where the selector returns the list of items and the component using the selector does in this case the join, I think makes a lot of sense 👍

File: src/components/data-model-table/association-selection.tsx [hunk 2]
```diff
@@ -131,173 +149,113 @@ export const AssociationSelection = <
...
-  } = useTable<GetQuery>({
-    ...tableOptions,
-  });
+  } = useTable<GetQuery>(tableOptions);
 
   const [selectedRows, setSelectedRows] = useState<AssociatedModel[]>(() =>
     value ? [value] : [],
   );
 
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (elliot-sabitov) [inline src/components/data-model-table/association-selection.tsx:142]: good to see that this section of code is no longer needed and we can plainToInstance elsewhere like in the reducer 👍

File: src/message/customer-information/index.tsx [hunk 3]
```diff
@@ -18,6 +21,11 @@ export const CustomerInformation: React.FC<GetCustomerProps> = ({
   };
 
   const getCustomersRequest = new CustomerInformationRequest();
+  const idToken = new IdToken();
+  idToken[IdTokenProps.id] = NEW_IDENTIFIER as any;
+  getCustomersRequest[GetCustomerInformationDataProps.clear] = false;
```
Comment (elliot-sabitov) [inline src/message/customer-information/index.tsx:26]: I dont think you need to set these two because booleans default to false automatically

Comment (deye9) [reply src/message/customer-information/index.tsx:26]: @elliot-sabitov it was defaulting to undefined for me which is why I set them as a default

**Review by thanaParis — APPROVED** [review]
Approved

**Review by elliot-sabitov — APPROVED** [review]

**Pull request merged**